src/object_detection.cpp

```diff
@@ -1,35 +1,39 @@
 #include <ros/ros.h>
-#include <sensor_msgs/Image.h>
-#include <object_detection_pkg/ObjDetected.h>
 #include <ros/console.h>
+#include <ros/package.h>
+#include <sensor_msgs/Image.h>
+#include <sensor_msgs/image_encodings.h>
 #include <geometry_msgs/Vector3.h>
-#include <fstream>
-#include <cstdlib>
 #include <std_msgs/String.h>
-#include "DetectedObject.h"
 #include <image_transport/image_transport.h>
-#include <cv_bridge/cv_bridge.h>
-#include <sensor_msgs/image_encodings.h>
-#include <opencv2/highgui/highgui.hpp>
+#include <object_detection_pkg/ObjDetected.h>
+#include <object_detection_pkg/DetectedObjsArray.h>
+#include <fstream>
+#include <cstdlib>
 #include <sstream>
 #include <iostream>
 #include <cstdio>
-// #include <nlohmann/json.hpp>
-
-// using json = nlohmann::json; // Define an alias for nlohmann::json
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <cv_bridge/cv_bridge.h>
+#include <opencv2/highgui/highgui.hpp>
+#include <jsoncpp/json/json.h>
+#include "DetectedObject.h"
 
+// Image_Finder class definition
 class Image_Finder {
 public:
-    
+    // Constructor
     Image_Finder() : nh_("") {
-        // Subscribe to camera topic publishing data
+        // Subscribing to a camera topic to receive image data
         sub_img_ = nh_.subscribe("/camera/color/image_raw", 1, &Image_Finder::imageCallback, this);
-        publisher = nh_.advertise<object_detection_pkg::ObjDetected>("/raw_image_copy", 1);
+        // Setting up a publisher for the detected objects
+        publisher = nh_.advertise<object_detection_pkg::DetectedObjsArray>("/Detected_Objects", 1);
```

Comment:
Naming convention for topics is lowercase with underscores: detected_objects/

Reply:
Done!